update release notes script to automatically grab labels from the PR

bdhirsh opened the pull request:
Now that developers are asked to add release note labels to their PR's, we can update the release notes scripts to look at those labels when it categorizes commits.

Tested by running `python commitlist.py --create_new be7177751eda6a86720d2e3d72c6660646e81ada  1659a267f992156f1b636274cb4a3f94455e7c05`, and seeing `dataloader,bug fixes` in the output (which came from a labeled [dataloader PR](https://github.com/pytorch/pytorch/pull/74958))

Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #75020
* #72154

bdhirsh committed:
update release notes script to automatically grab labels from the PR

[ghstack-poisoned]

albanD reviewed (approved):
👍

bdhirsh committed:
Update on "update release notes script to automatically grab labels from the PR"

Now that developers are asked to add release note labels to their PR's, we can update the release notes scripts to look at those labels when it categorizes commits.

Tested by running `python commitlist.py --create_new be7177751eda6a86720d2e3d72c6660646e81ada  1659a267f992156f1b636274cb4a3f94455e7c05`, and seeing `dataloader,bug fixes` in the output (which came from a labeled [dataloader PR](https://github.com/pytorch/pytorch/pull/74958))




[ghstack-poisoned]

anjali411 commented on scripts/release_notes/commitlist.py:
```diff
@@ -84,15 +84,29 @@ def keywordInFile(file, keywords):
+        already_categorized = already_topiced = False
+        for label in labels:
+            if label.startswith('release notes: '):
+                category = label.split('release notes: ', 1)[1]
```
are the categories used by the script in line with the category labels?

anjali411 commented on scripts/release_notes/commitlist.py:
```diff
@@ -84,15 +84,29 @@ def keywordInFile(file, keywords):
-        labels = features['labels']
         if 'Reverted' in labels:
             return Commit(commit_hash, 'skip', topic, title)
         if 'bc_breaking' in labels:
```
`module: deprecation` is not a topic, so we should change it to 'deprecation'

bdhirsh committed:
Update on "update release notes script to automatically grab labels from the PR"

Now that developers are asked to add release note labels to their PR's, we can update the release notes scripts to look at those labels when it categorizes commits.

Tested by running `python commitlist.py --create_new be7177751eda6a86720d2e3d72c6660646e81ada  1659a267f992156f1b636274cb4a3f94455e7c05`, and seeing `dataloader,bug fixes` in the output (which came from a labeled [dataloader PR](https://github.com/pytorch/pytorch/pull/74958))




[ghstack-poisoned]

bdhirsh committed:
Update on "update release notes script to automatically grab labels from the PR"

Now that developers are asked to add release note labels to their PR's, we can update the release notes scripts to look at those labels when it categorizes commits.

Tested by running `python commitlist.py --create_new be7177751eda6a86720d2e3d72c6660646e81ada  1659a267f992156f1b636274cb4a3f94455e7c05`, and seeing `dataloader,bug fixes` in the output (which came from a labeled [dataloader PR](https://github.com/pytorch/pytorch/pull/74958))




[ghstack-poisoned]

bdhirsh commented:
@pytorchbot merge this please

bdhirsh commented:
@pytorchbot merge this please

bdhirsh committed:
Update on "update release notes script to automatically grab labels from the PR"

Now that developers are asked to add release note labels to their PR's, we can update the release notes scripts to look at those labels when it categorizes commits.

Tested by running `python commitlist.py --create_new be7177751eda6a86720d2e3d72c6660646e81ada  1659a267f992156f1b636274cb4a3f94455e7c05`, and seeing `dataloader,bug fixes` in the output (which came from a labeled [dataloader PR](https://github.com/pytorch/pytorch/pull/74958))




[ghstack-poisoned]

anjali411 commented:
@bdhirsh you probably want to land the PR that's below this in the stack right?

bdhirsh commented:
yep - I might have the wrong incantation, but I thought "@pytorchbot merge this please" is supposed to land the whole stack if you run it on the top PR

bdhirsh commented:
@pytorchbot merge this please
